### PR TITLE
enable_qwen2-vl_quantization

### PR DESCRIPTION
**Pull request by WeiweiZhang1**

### CODE REVIEW

File: auto_round/autoround.py [hunk 1]
```diff
@@ -939,10 +966,11 @@ def quant_block(self, block, input_ids, input_others, q_input=None, device=torch
                     seqlen=self.seqlen,
                     share_attention_mask_flag=self.share_attention_mask_flag,
                     not_share_position_ids_flag=self.not_share_position_ids_flag,
+                    not_share_rotary_pos_emb_flag=self.not_share_rotary_pos_emb_flag,
                     input_dim=self.input_dim,
                 )
 
-                current_output = [output[i] for i in indices]
+                current_output = [output[x] for x in indices]
```
Comment (wenhuach21) [inline auto_round/autoround.py:973]: why use x

File: auto_round/autoround.py [hunk 2]
```diff
@@ -690,13 +705,25 @@ def forward(m, hidden_states, *positional_args, **kwargs):
...
-                            self.inputs[name][key] = list(torch.split(kwargs[key].to("cpu"), 1, dim=0)) \
-                                if self.not_share_position_ids_flag \
-                                else to_device(kwargs[key], device=torch.device("cpu"))
+                    elif "position_ids" in key or 'cache_position' in key:
+                        if self.train_bs == 1 and self.not_share_rotary_pos_emb_flag:
+                            if key not in self.inputs[name].keys():
+                               self.inputs[name][key] = [to_device(kwargs[key], device=torch.device("cpu"))]
+                            else:
+                                self.inputs[name][key].append(to_device(kwargs[key], device=torch.device("cpu")))
+                        elif key not in self.inputs[name].keys():
```
Comment (wenhuach21) [inline auto_round/autoround.py:714]: The code is getting more and more complicated, which is hard to maintain

**Review by wenhuach21 — APPROVED** [review]

File: auto_round/autoround.py (hunk 1 above)
Comment (WeiweiZhang1) [reply auto_round/autoround.py:973]: To distinguish it from the index i of the for loop and avoid potential problems

File: auto_round/autoround.py (hunk 2 above)
Comment (WeiweiZhang1) [reply auto_round/autoround.py:714]: will refine multimodal code later

File: examples/language-modeling/main.py [hunk 3]
```diff
@@ -94,9 +94,9 @@
     parser.add_argument("--enable_minmax_tuning", action='store_true',
                         help="enable_minmax_tuning is deprecated")
 
-    parser.add_argument("--deployment_device", default='auto_round', type=str,
-                        help="targeted inference acceleration platform,The options are 'fake', 'cpu', 'xpu', 'gpu(auto_gptq)' and 'auto_round'."
-                             "default to 'auto_round', 'fake' indicating that it only performs fake quantization and won't be exported to any device.")
+    parser.add_argument("--deployment_device", default=None, type=str,
+                        help="targeted inference acceleration platform,The options are 'fake', 'cpu', 'gpu' and 'xpu'."
+                             "default to 'fake', indicating that it only performs fake quantization and won't be exported to any device.")
```
Comment (wenhuach21) [inline examples/language-modeling/main.py]: default should be auto_round

Comment (wenhuach21) [reply examples/language-modeling/main.py]: help revert this

**Pull request merged**